composer/callbacks/checkpoint_saver.py

```diff
@@ -301,119 +301,117 @@ def __init__(
         weights_only: bool = False,
     ):
         if not callable(save_interval):
-            save_interval = checkpoint_periodically(save_interval)
+            self.save_interval = checkpoint_periodically(save_interval)
```

Comment:
if I pass `save_interval` as a callable, I think this change sets it to `None` now. I believe we need `self.save_interval = save_interval` outside the if statement